packages/ic-management/src/ic-management.canister.ts

```diff
@@ -1,60 +1,178 @@
-import {
-  getManagementCanister,
-  type ManagementCanisterRecord,
-} from "@dfinity/agent";
 import { Principal } from "@dfinity/principal";
-import { toNullable } from "@dfinity/utils";
+import { createServices, toNullable } from "@dfinity/utils";
+import type { _SERVICE as IcManagementService } from "../candid/ic-management";
+import { idlFactory as certifiedIdlFactory } from "../candid/ic-management.certified.idl";
+import { idlFactory } from "../candid/ic-management.idl";
 import type { ICManagementCanisterOptions } from "./types/canister.options";
-import type { UpdateSettingsParams } from "./types/ic-management.params";
-import type { CanisterStatusResponse } from "./types/ic-management.responses";
+import type {
+  CanisterInfoParams,
+  CreateCanisterParams,
+  UninstallCodeParams,
+} from "./types/ic-management.params";
+import {
+  toCanisterSettings,
+  toInstallMode,
+  type InstallCodeParams,
+  type UpdateSettingsParams,
+} from "./types/ic-management.params";
+import type {
+  CanisterInfoResponse,
+  CanisterStatusResponse,
+} from "./types/ic-management.responses";
 
 export class ICManagementCanister {
-  private constructor(private readonly service: ManagementCanisterRecord) {
+  private constructor(private readonly service: IcManagementService) {
     this.service = service;
   }
 
   public static create(options: ICManagementCanisterOptions) {
-    const agent = options.agent;
-
-    const service = options.serviceOverride ?? getManagementCanister({ agent });
+    const { service } = createServices<IcManagementService>({
+      options: {
+        ...options,
+        canisterId: Principal.fromHex(""),
```

Comment:
What's the result of `Principal.fromHex("")` ?  `aaaaa-aa`?

Reply:
Yes it's `aaaaa-aa`. I've updated it to use `fromText('aaaaa-aa')` instead to make it easier more obvious.